core/src/main/java/org/geysermc/geyser/GeyserImpl.java

```diff
@@ -308,15 +308,17 @@ private void startInstance() {
                 }
             }
 
-            boolean floodgatePresent = bootstrap.testFloodgatePluginPresent();
-            if (config.getRemote().authType() == AuthType.FLOODGATE && !floodgatePresent) {
-                logger.severe(GeyserLocale.getLocaleStringLog("geyser.bootstrap.floodgate.not_installed") + " "
-                        + GeyserLocale.getLocaleStringLog("geyser.bootstrap.floodgate.disabling"));
-                return;
-            } else if (config.isAutoconfiguredRemote() && floodgatePresent) {
-                // Floodgate installed means that the user wants Floodgate authentication
-                logger.debug("Auto-setting to Floodgate authentication.");
-                config.getRemote().setAuthType(AuthType.FLOODGATE);
+            if (platformType != PlatformType.VIAPROXY) {
```

Comment:
Instead of adding a viaproxy specific check here, couldnt you just override the `testFloodgatePluginPresent` method in the viaproxy bootstrap? The same is done in standalone aswell.

Reply:
I didn't want to return true in that method since it might have some side effects when the method gets used for something else as it implies the platform has the Floodgate plugin installed. I just added support for joining servers which have Floodgate installed and the if check was needed so Geyser doesn't complain that the Floodgate plugin isn't installed on ViaProxy itself

Reply:
I'd recommend to check how Geyser Standalone does it - there's also no Floodgate for Standalone, but Floodgate auth still works with Standalone if it's installed on the server one is connecting to

Reply:
Floodgate auth works in Standalone, because the user can configure it in the config. If the user would configure it in ViaProxy it would cause an exception since Geyser checks if the Floodgate plugin is installed when Floodgate auth is enabled (There is an if check (https://github.com/GeyserMC/Geyser/blob/4e490634a3a018375a2e29da2f60912aca76a8ed/core/src/main/java/org/geysermc/geyser/GeyserImpl.java#L255) which skips this and the remote auto configuring if the platform is Standalone, but I use the remote auto configuring in ViaProxy, thats why I had to introduce a new if check (https://github.com/GeyserMC/Geyser/blob/4e490634a3a018375a2e29da2f60912aca76a8ed/core/src/main/java/org/geysermc/geyser/GeyserImpl.java#L311) which only skips the Floodgate check)

Reply:
I see, thanks for the explainer :)